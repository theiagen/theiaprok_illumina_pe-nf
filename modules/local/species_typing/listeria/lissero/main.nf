process LISSERO {
    tag "$meta.id"
    label "process_low"

    container "us-docker.pkg.dev/general-theiagen/biocontainers/lissero:0.4.9--py_0"

    input:
    tuple val(meta), path(assembly)
    val min_percent_identity
    val min_percent_coverage

    output:
    tuple val(meta), path("*.tsv"), emit: lissero_results
    tuple val(meta), path("lissero_SEROTYPE.txt"), emit: lissero_serotype
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def min_id_arg = min_percent_identity ? "${min_percent_identity}" : 95.0
    def min_cov_arg = min_percent_coverage ? "${min_percent_coverage}" : 95.0 
    
    """
    lissero \\
        --min_id ${min_id_arg} \\
        --min_cov ${min_cov_arg} \\
        ${assembly} \\
        ${args} \\
        > ${prefix}.tsv

    # pull out serotype
    if [ -f ${prefix}.tsv ] && [ \$(wc -l < ${prefix}.tsv) -gt 1 ]; then
        serotype=\$(tail -n+2 ${prefix}.tsv | cut -f2)
    else
        serotype="No serotype predicted"
    fi
    echo "\$serotype" > lissero_SEROTYPE.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        lissero: \$(lissero --version 2>&1 | sed 's/^.*LisSero //')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.tsv
    echo "No serotype predicted" > lissero_SEROTYPE.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        lissero: \$(lissero --version 2>&1 | sed 's/^.*LisSero //')
    END_VERSIONS
    """
}