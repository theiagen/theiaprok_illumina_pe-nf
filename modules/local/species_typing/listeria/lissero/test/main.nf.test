nextflow_process {
    name "Test Process LISSERO"
    script "../main.nf"
    process "LISSERO"

    test("LISSERO - test") {
        setup {
            run("GUNZIP") {
                script "../../../../../nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = [
                        [id:'test_sample'],
                        [file('https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/848/505/GCF_002848505.1_ASM284850v1/GCF_002848505.1_ASM284850v1_genomic.fna.gz', checkIfExists: true)]
                    ]
                    """
                }
            }
        }
        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = 90.0
                input[2] = 90.0
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.lissero_results,
                    process.out.lissero_serotype,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("lissero - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/escherichia_coli/genome/genome.fa.gz', checkIfExists: true)
                ])
                input[1] = 90.0
                input[2] = 95.0
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}