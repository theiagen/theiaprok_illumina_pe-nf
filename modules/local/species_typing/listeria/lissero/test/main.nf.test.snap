{
    "LISSERO - test": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample.tsv:md5,028387ad8c00f7511c78416cba9999f0"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "SEROTYPE.txt:md5,db31935632c995351cf6cd5d0ca52772"
                ]
            ],
            [
                "versions.yml:md5,f091db71715241a5f46beacaf97dc08b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-24T21:27:57.21742011"
    },
    "lissero - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "SEROTYPE.txt:md5,d4b61d77cb0b810024447774a7d4d141"
                    ]
                ],
                "2": [
                    "versions.yml:md5,f091db71715241a5f46beacaf97dc08b"
                ],
                "lissero_results": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "lissero_serotype": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "SEROTYPE.txt:md5,d4b61d77cb0b810024447774a7d4d141"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f091db71715241a5f46beacaf97dc08b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-24T21:28:10.664525911"
    }
}